### PR TITLE
Start adding basic Vue stuff to debug through environment issues

### PR DESCRIPTION
**Pull request by sarayourfriend**
cc @visualjerk

I'm opening this PR to start working through the environment issues with adding Vue related libraries into Ariakit.

I am beginning to think that it might not be possible to do this, but I need to dig into it a little more. Right now if you run `npm run type-check` you'll see a bunch of issues that essentially amount to Vue and React types fighting each other. Curiously this time it seems that Vue is trashing the React types:

```
packages/website/components/playground/playground-error.tsx:9:5 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Omit<CommandOptions<"button">, "as"> & Omit<HTMLProps<Options<"pre">>, "as" | "autoFocus" | "disabled" | "focusable" | "accessibleWhenDisabled" | "onFocusVisible" | "clickOnEnter" | "clickOnSpace"> & Required<...>): Element | null', gave the following error.
    Type '{ focusable: boolean; onClick: () => void; className: string | undefined; as: "pre"; disabled?: boolean | undefined; autoFocus?: boolean | undefined; accessibleWhenDisabled?: boolean | undefined; ... 257 more ...; onTransitionEndCapture?: TransitionEventHandler<...> | undefined; }' is not assignable to type 'IntrinsicAttributes'.
      Types of property 'ref' are incompatible.
        Type '((instance: HTMLPreElement | null) => void) | RefObject<HTMLPreElement> | null | undefined' is not assignable to type 'VNodeRef | undefined'.
          Type 'null' is not assignable to type 'VNodeRef | undefined'.
  Overload 2 of 2, '(props: Props<CommandOptions<"button">>): Element | null', gave the following error.
    Type '"pre"' is not assignable to type '"button"'.

  9     <Command
        ~~~~~~~~
 10       as="pre"
    ~~~~~~~~~~~~~~
... 
 26       }
    ~~~~~~~
 27     />
    ~~~~~~
```

There may be some way to fix this by moving the TypeScript configuration into separate React and Vue ones, like I did with babel. I'll try that next :crossed_fingers:

### CODE REVIEW

**Comment by diegohaz** [comment]
Does it work if we install `@types/react` and Vue types in the specific packages (without hoisting — not sure if this works automatically)?

File: packages/ariakit-react-utils/package.json [hunk 1]
```diff
@@ -33,6 +33,9 @@
     "react",
     "utils"
   ],
+  "dependencies": {
+    "ariakit-utils": "0.17.0-next.26"
+  },
```
Comment (diegohaz) [inline packages/ariakit-react-utils/package.json:38]: Nice 👍 
Could you do that in separate PR, please?

Comment (sarayourfriend) [reply packages/ariakit-react-utils/package.json:38]: Yup, sure thing!

**Comment by sarayourfriend** [comment]
> Does it work if we install @types/react and Vue types in the specific packages (without hoisting — not sure if this works automatically)?

Good idea. This had crossed my mind but then I'd forgotten about it. I'll try it out!

**Comment by sarayourfriend** [comment]
Actually, looking again and I don't think it will work. I'm going to try it just in case, but my hypothesis is that because currently the whole workspace is treated as a single TypeScript project, the React types and the Vue types are going to both get included into the project. This `include` line in particular is of interest to me: https://github.com/ariakit/ariakit/blob/HEAD/tsconfig.prod.json#L25
